### ISSUE TITLE
Embeddings exist for entities that do not have pages in Wikipedia

### ISSUE THREAD

**Issue by katlap** [wikipedia2vec/wikipedia2vec#57]
While using Wikipedia2Vec (which is an amazing tool by the way, thanks so much for making it available!) I occasionally find that model.get_entity_vector gives embeddings for the titles that do not have corresponding pages in Wikipedia. For example, I can retrieve embedding vectors for "Cultural awareness"  or "Business communications", but https://en.wikipedia.org/wiki/Cultural_awareness page does not exist. I found a wikidata item for "Cultural awareness", but nothing for "Business communications".  Does this mean that the Wikipedia dump I used for training the model contained information on these pages, but since then they were removed from Wikipedia? Thank you very much!

**Comment by ikuyamada** [comment]
Thank you for using Wikipedia2Vec! The current implementation of Wikipedia2Vec does not check whether the entity is a valid one. If Wikipedia contains hyperlinks referring to the entity more than `--min-entity-count` times, the entity is included in the model.

**Comment by katlap** [comment]
Thank you so much for your prompt answer! So, as I understand correctly, the hyperlink doesn't have to be active for the title to be an "entity"? Also, would you say that if there is no page associated with the entity, the corresponding embedding may not have as high confidence as there was less context available for training that entity?

**Comment by ikuyamada** [comment]
> Also, would you say that if there is no page associated with the entity, the corresponding embedding may not have as high confidence as there was less context available for training that entity?

I believe that there are not many cases that Wikipedia hyperlinks point to inexistent entities. the resulting embeddings of such entities can be poor.

**Comment by katlap** [comment]
Yes, I find this in <1% cases which is not too bad. Thank you for your help!

**Comment by johannesemme** [comment]
I would also like to thank you very much for sharing this great work, which I am currently using for trying to reproduce the results from "Joint Learning of the Embedding of Words and Entities for Named Entity Disambiguation" for my master's thesis. 

I am, however, very much curious about the conclusion of this issue :) 

I have recently downloaded the latest English dump (8.622.133 titles)
From these titles I have retrieved the non-categories to obtain 6.470.841 articles (1.873 less than reported at 
https://en.wikipedia.org/wiki/Wikipedia:Size_of_Wikipedia which seems fine)

Next, I have created a dictionary with settings: 
--min-word-count 5 --min-entity-count 2 --min-paragraph-len 5
Hereby I obtain:
5.733.103 entities
2.196.922 words

However, I find that the overlap of entities between the dictionary and the articles from the dump is only 4.604.203 (~80%).

When examining the ~20% non-overlapping entities between the dictionary and wiki articles, I find that these entities mainly:
1) Do not have pages in Wikipedia (as described above)
2) Consist of a bunch of unwanted image files, category pages, wiktionary pages etc.

I tried to remove some of these unwanted files with this (non comprehensive) regex: 
ENTS_IGNORE = re.compile(r"\s*(file|image|wikt|wikipedia|wiktionary|category|commons|template|iarchive|wikisource|scores|user|special|portal):", re.IGNORECASE)

This however still leaves me with ~740.000 entities that (mostly) seems to be cases of Wikipedia hyperlinks that point to nonexisting Wikipedia articles (entities). 

Long story short: I find that the problem of hyperlinks pointing to nonexisting entities is much more prominent.
